food-truck.eve.md

```diff
@@ -352,19 +352,118 @@ bind @browser
 - If either name or price forms do not have information, pop up an error message saying “You’re missing a name/price!” with an OK button to revert to item listing
 
 ### Social Media
-- Draw top banner
-- Draw truck name
-- Draw back button
-- Clicking takes you back to the home screen
-- Draw message text form
-- Draw picture icon
-- Clicking lets you choose photo from phone storage or live photo
-- Draw social media icons
-- Facebook
-- Twitter
-- Instagram
-- All off by default
-- Clicking turns on that site for posting
+
+Draw the social media page. There are two cases we want to handle. If the truck isn't set up or there are no #enabled social media #integrations, then we should display a message prompting the owner to add those in.
+
+```
+search @browser @session
+  wrapper = [#page-wrapper page: "social-flow"]
+  // Is there a better way to do this? I want the message to display if either of these conditions are true
+  x = if not([#app settings: [truck-name]]) then ""
+      if not([#integration #enabled]) then ""
+  
+bind @browser
+  wrapper.children := [#div children: 
+    [#div text: "Set up your truck to use social features"]
+    [#button #to-settings text: "Truck Settings"]]
+```
+
+Navigate to the settings page when the button is clicked
+
+```
+search
+  app = [#app]
+  
+search @event @browser
+  [#click element: [#button #to-settings]]
+  
+commit
+  app.page := "settings"
+```
+
+When the truck has a name and at least one integration, draw the complete social interface
+
+```
+search @browser
+  wrapper = [#page-wrapper page: "social-flow"]
+  
+search
+  app = [#app]
+  name = if app.settings.truck-name then app.settings.truck-name
+         else "Default"
+
+  // Integrations
+  integration = [#integration #enabled]
+  selected? = if integration = [#selected] then ""
+              else "-outline"
+  
+bind @browser
+  wrapper.children := [#div children: 
+    [#div text: name]
+    [#button #to-home text: "home"]
+    [#div children:
+      [#div text: "Add a message:"]
+      [#editable #social-message default: "message..."]]    
+    [#div #integrations children:
+      [#div text: "Add a photo:"]
+      [#image-container #social-image prompt: "photo..."]]
+    [#div children: 
+      [#div text: "Select social networks"]
+      [#span #integration integration class: "ion-social-{{integration.name}}{{selected?}}"]]
+    [#button #post-social text: "Post"]]
+```
+
+Clicking an integration toggles its selection status for posting
+
+```
+search
+  [#app page: "social-flow"]
+
+search @browser @event @session
+  [#click element]
+  element = [#integration integration]
+  (add, remove) = if integration = [#selected] then ("","selected")
+                  else ("selected", "")
+  
+  
+commit
+  integration.tag += add
+  integration.tag -= remove  
+```
+
+The "post" button is enabled when at least one social integration is selected, and a message or an image is uploaded
```

Comment:
I add a class when it's enabled, which should provide the visual indication when the app is styled.